packages/opentelemetry-instrumentation-google-generativeai/opentelemetry/instrumentation/google_generativeai/span_utils.py

```
@@ -225,217 +226,89 @@ async def upload_task():
-        for p in content_item.get("parts", []):
-            if isinstance(p, dict):
-                parts_acc.append(_map_dict_part(p))
-            else:
-                parts_acc.extend(await _parts_from_genai_part_async(p, span, 0))
-    elif hasattr(content_item, "parts"):
+    """Process a single content item, handling different types (Content objects, strings, Parts)"""
+    processed_content = []
+    if hasattr(content_item, "parts"):
+        # Content with parts (Google GenAI Content object)
         for part_index, part in enumerate(content_item.parts):
             parts_acc.extend(await _parts_from_genai_part_async(part, span, part_index))
```

Ruff (0.15.9)

[error] Undefined name `parts_acc`

(F821)